habitat_sim/utils/viz_utils.py

```diff
@@ -26,6 +31,260 @@ def is_notebook():
+        sensor_spec.orientation = sensor_params["orientation"]
+        sensor_specs.append(sensor_spec)
+
+    # agent configurations
+    agent_cfg = habitat_sim.agent.AgentConfiguration()
+    agent_cfg.sensor_specifications = sensor_specs
+
+    return habitat_sim.Configuration(backend_cfg, [agent_cfg])
+
+
+def make_video(
+    observations,
+    primary_pov,
+    output_path,
+    file_name,
+    camera_res=[540, 720],
+    open_vid=True,
+    multi_obs=False,
+    pov_list=[],
+):
+    """
+    Build a video of passed observations array, with embedded images if desired
+
+    Parameters
+    ----------
+    observations : list
+        List of observations from which the video should be constructed
+
+    primary_pov : string
+        Primary camera name in observations to be used for image.
+
+    output_path : string
+        location where to save resultant .mp4 video.
+
+    file_name : string
+        File name to use to save resultant .mp4 video.
+
+    camera_res : list, optional
+        Height by Width of camera for image. Default is [540, 720]
+
+    open_vid : boolean, optional
+        Whether to open video upon creation.  Default is True
+
+    multi_obs : boolean, optional
+        Whether to embed images from POVs listed in pov_list.
+        Defaults to False
+
+    pov_list : List of tuples, optional
+        If present, each tuple holds :
+            idx 0 is camera name in observations
+            idx 1 is boolean whether is depth canera or not.
+        List of strings denoting specific camera types and POVs to consume
+        from observations list.  Ignored if multi_obs is False
+
+    Returns
+    -------
+    None.
+
+    """
+    videodims = (camera_res[1], camera_res[0])
+    video_file = output_path + file_name + ".mp4"
+    print("Encoding the video: %s " % video_file)
+    writer = get_fast_video_writer(video_file, fps=60)
+    thumb_size = (int(videodims[0] / 5), int(videodims[1] / 5))
+    outline_frame = np.ones((thumb_size[1] + 2, thumb_size[0] + 2, 3), np.uint8) * 150
+
+    if multi_obs and len(pov_list) > 1:
+        embed_image_list = []
+        for i in range(0, len(pov_list)):
+            # don't place primary POV in embedded image list
+            if pov_list[i][0].lower() in primary_pov.lower():
+                continue
+            embed_image_list.append((pov_list[i][0], "depth" in pov_list[i][1].lower()))
+
+        for ob in observations:
+            # build list of embedded POV images and whether or not they are depth
+            embed_image_data_list = [
+                (ob[embed_image_list[i][0]], embed_image_list[i][1])
+                for i in range(0, len(embed_image_list))
+            ]
+
+            res_image = build_multi_obs_image(
+                ob[primary_pov], embed_image_data_list, thumb_size, outline_frame,
+            )
+            if res_image.shape[:2] != videodims:
+                res_image = cv2.resize(
```

Comment:
I know we are running a bit late on this, so it's low pri, but we can refactor this to PIL/numpy and remove the CV2 dependency entirely.

Reply:
Sounds good.  I'll address these issues as soon as I can.